src/models/Projection.ts

```diff
@@ -49,98 +28,47 @@ export class Projection {
-      _parseInt(row[COLUMNS.hospitalizations]),
-    );
-    this.beds = data.map(row => _parseInt(row[COLUMNS.beds]));
-    this.deaths = data.map(row => _parseInt(row[COLUMNS.deaths]));
-    this.cumulativeDeaths = this.deaths;
-    this.totalPopulation = _parseInt(data[0][COLUMNS.totalPopulation]);
-    this.infected = data.map(row => _parseInt(row[COLUMNS.infected]));
-
-    this.cumulativeInfected = [];
-    let infectedSoFar = 0;
-    for (let i = 0; i < this.infected.length; i++) {
-      infectedSoFar += this.infected[i];
-      this.cumulativeInfected.push(infectedSoFar * (2 / 3));
-    }
+    this.hospitalizations = timeseries.map(row => row.hospitalBedsRequired);
+    this.beds = timeseries.map(row => row.hospitalBedCapacity);
+    this.cumulativeDeaths = timeseries.map(row => row.cumulativeDeaths);
+    this.totalPopulation = summaryTimeseries.actuals.population;
+    this.cumulativeInfected = timeseries.map(row => row.cumulativeInfected);
 
     this.cumulativeDead = this.cumulativeDeaths[
       this.cumulativeDeaths.length - 1
     ];
 
-    // approximate the date when the hospitals will be overwhelmed
-    // assume r0~=2 => 25% daily case growth
-    let overwhelmedIdx = this.dates.findIndex(
-      (num, idx) => this.hospitalizations[idx] > this.beds[idx],
-    );
-
-    if (overwhelmedIdx === -1) {
-      this.dateOverwhelmed = null;
-    } else {
-      // since we only get reports every 4 daysfind x coordinate of the crossing point of
-      // hospitalizations and beds.
-      const startIdx = Math.max(0, overwhelmedIdx - 1); // can't get overwhelmed before day 0
-      const endIdx = overwhelmedIdx;
-      // create coords normalized with an x range of 1
-      const bedsStart = { y: this.beds[startIdx], x: 0 };
-      const bedsEnd = { y: this.beds[endIdx], x: 1 };
-      const hospitalizationsStart = {
-        y: this.hospitalizations[startIdx],
-        x: 0,
-      };
-      const hospitalizationsEnd = {
-        y: this.hospitalizations[endIdx],
-        x: 1,
-      };
-      // find the point at which they intersect. this method uses linear interpolation for
-      // simplicity so there may be some disconnect here.
-      const crossingPoint = intersection(
-        bedsStart,
-        bedsEnd,
-        hospitalizationsStart,
-        hospitalizationsEnd,
-      );
-      // scale the normalized range of 4 back up to the actual range of 4 days.
-      let dayDelta = Math.floor(4 * crossingPoint.x);
-      // create the new date by adding the delta to the day prior to overwhelming.
-      const startDate = this.dates[startIdx];
-      const deltaSecs = dayDelta * MS_PER_DAY;
-      const estimatedTimeStamp = startDate.getTime() + deltaSecs;
-      this.dateOverwhelmed = new Date(estimatedTimeStamp);
-    }
+    const shortageStart =
+      summaryTimeseries.projections.totalHospitalBeds.shortageStartDate;
+    this.dateOverwhelmed =
+      shortageStart === null ? null : new Date(shortageStart);
```

Comment:
this will be funky until we switch to datapoint per day (the old code was interpolating between the points) but since we're switching, seems fine!

Reply:
Yep, agreed.  That's next on my list.